File: src/de/fssd/dataobjects/MCState.java
package de.fssd.dataobjects;

import com.sun.istack.internal.NotNull;

import java.util.List;

public class MCState implements Comparable {
    private final @NotNull String id;
    private final float p0;
    private final @NotNull List<String> out;
    private final @NotNull List<MCTransition> transitions;

    public MCState(String id, float p0, List<String> out, List<MCTransition> transitions) {
        this.id = id;
        this.p0 = p0;
        this.out = out;
        this.transitions = transitions;
    }

    public String getId() {
        return id;
    }

    public List<String> getOut() {
        return out;
    }

    public float getP0() {
        return p0;
    }

    public List<MCTransition> getTransitions() {
        return transitions;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        MCState mcState = (MCState) o;

        return getId().equals(mcState.getId());

    }

    @Override
    public int hashCode() {
        return getId().hashCode();
    }

    @Override
    public String toString() {
        return "MCState{" +
                "id='" + id + '\'' +
                ", p0=" + p0 +
                ", out=" + out +
                '}';
    }

    @Override
    public int compareTo(Object o) {
        if(o instanceof MCState) {
            return this.id.compareTo(((MCState) o).id);
        } else {
            return 0;
        }
    }
}

File: src/de/fssd/dataobjects/MCState.kt
package de.fssd.dataobjects

class MCState(val id: String, val p0: Float, val out: List<String>, val transitions: List<MCTransition>) : Comparable<MCState> {



    override fun hashCode(): Int {
        return id.hashCode()
    }

    override fun toString(): String {
        return "MCState{" +
                "id='" + id + '\'' +
                ", p0=" + p0 +
                ", out=" + out +
                '}'
    }

    override fun compareTo(other: MCState): Int {
        return id.compareTo(other.id)
    }

    override fun equals(other: Any?): Boolean{
        if (this === other) return true
        if (other !is MCState) return false

        if (id != other.id) return false

        return true
    }
}
